Cargo.toml

```diff
@@ -107,6 +107,7 @@ cluster = { path = "src/cluster" }
 criterion = "0.5"
 horaedb-client = "1.0.2"
 common_types = { path = "src/common_types" }
+dashmap = "5.5.3"
```

Comment:
I wonder if this is a required dependency for this task?

If HashMap works, I prefer to stick with it first.

Reply:
I'm trying to run 'replay_table_log_entries' concurrently, but I faced an issue with 'serial_exec_ctxs', which is a mutable reference HashMap. I had to wrap this by Arc and Mutex, then every time I grap a mutable reference to a value from the map, it locks the entire map. 

DashMap allowing concurrent access to different keys. I wonder if there's an appoach to make Hashmap work in this case.

Reply:
> then every time I grap a mutable reference to a value from the map, it locks the entire map.


I think it's find to use plain Mutex here, since they are not the bottle neck, `replay_table_log_entries` is the most heavy task in this place.

Also, there is a partitioned lock in our codebase, you can use if you want to optimize here:
https://github.com/apache/incubator-horaedb/blob/9f166f3daa9a02ef8af1e733c22f956ab97e7aaf/src/components/partitioned_lock/src/lib.rs#L130

Reply:
> > then every time I grap a mutable reference to a value from the map, it locks the entire map.
> 
> I think it's find to use plain Mutex here, since they are not the bottle neck, `replay_table_log_entries` is the most heavy task in this place.
> 
> Also, there is a partitioned lock in our codebase, you can use if you want to optimize here:
> 
> https://github.com/apache/incubator-horaedb/blob/9f166f3daa9a02ef8af1e733c22f956ab97e7aaf/src/components/partitioned_lock/src/lib.rs#L130

Awesome! I'm gonna check it out. Thx for the heads up